homeassistant/components/imeon_inverter/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Config flow for Imeon integration."""
+
+import logging
+from typing import Any
+
+from imeon_inverter_api.inverter import Inverter
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ADDRESS, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ADDRESS): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ImeonInverterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the initial setup flow for Imeon Inverters."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the user step for creating a new configuration entry."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            connection = False
+
+            async with Inverter(user_input[CONF_ADDRESS]) as client:
+                try:
+                    # Check connection
+                    connection = await client.login(
+                        user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                    )
```

Comment:
```suggestion
                    if await client.login(
                        user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
                    ):
                         serial = await client.get_serial()
```
just an idea to make the code a bit more concise, since you'll only need one try and they are throwing the same Timeouterror